### DISCUSSION TITLE
Can I have a unique AsyncNotifier for each note in a note taking app?

### DISCUSSION THREAD

**Discussion by microcoder-py** [rrousselGit/riverpod#2924] [Q&A]
Assume I have a note taking app. There are folders, and each folder holds a set of notes. Currently, I have a small number of notes so I am using a single AsyncNotifier class to maintain them all instead of using a family. What I want to do instead is provide a unique provider to each Note so any changes to state only triggers rebuilds for that specific note 

Is this the right way to do it or should I rely upon family? And if I use family, how can I declare a class where I can hold all functions instead of using providers defined externally?

**Comment by rrousselGit** [comment]
The right way to do it is to use family, yes.
That's what it does. It creates a unique provider per unique parameter.
